src/QsCompiler/CompilationManager/AssemblyLoader.cs

```diff
@@ -172,6 +160,79 @@ public static bool LoadReferencedAssembly(Uri asm, out References.Headers header
             return compilation != null && IsValidCompilation(compilation);
         }
 
+        /// <summary>
+        /// Loads any QIR byte code included as a resource from <paramref name="assemblyFileInfo"/>.
+        /// </summary>
+        /// <param name="assemblyFileInfo">The file info of a .NET DLL from which to load the QIR byte code.</param>
+        /// <param name="qirByteStream">A stream for reading the QIR byte code included as a resource of <paramref name="assemblyFileInfo"/>.</param>
+        /// <returns>
+        /// True if <paramref name="assemblyFileInfo"/> includes the QIR byte code resource, false otherwise.
+        /// </returns>
+        /// <exception cref="FileNotFoundException"><paramref name="assemblyFileInfo"/> does not exist.</exception>
+        public static bool LoadQirByteCode(FileInfo assemblyFileInfo, Stream qirByteStream)
+        {
+            using var assemblyFile = GetAssemblyReader(assemblyFileInfo.FullName);
+            return LoadResource(DotnetCoreDll.ResourceNameQsDataQirV1, assemblyFile, qirByteStream);
+        }
+
+        /// <summary>
+        /// Gets a Portable Executable Reader object for the file specified by the path <paramref name="assemblyPath"/>.
+        /// </summary>
+        /// <param name="assemblyPath">The path to the file.</param>
+        /// <returns>
+        /// The PEReader for the given assembly.
+        /// </returns>
+        /// <exception cref="FileNotFoundException"><paramref name="assemblyPath"/> does not exist.</exception>
+        private static PEReader GetAssemblyReader(string assemblyPath)
+        {
+            if (!File.Exists(assemblyPath))
+            {
+                throw new FileNotFoundException($"The file '{assemblyPath}' given to the assembly loader does not exist.");
+            }
+            return new PEReader(File.OpenRead(assemblyPath));
+        }
+
+        /// <summary>
+        /// Loads a resource named <paramref name="resourceName"/> from <paramref name="assemblyReader"/> into the stream <paramref name="resourceStream"/>.
+        /// </summary>
+        /// <param name="resourceName">The name of the resource to load.</param>
+        /// <param name="assemblyReader">The Portable Executable Reader for the dll from which to load the resource.</param>
+        /// <param name="resourceStream">A stream for reading the resource as a bytes.</param>
+        /// <returns>
+        /// True if <paramref name="assemblyReader"/> includes the resource named by <paramref name="resourceName"/>, false otherwise.
+        /// </returns>
+        private static bool LoadResource(string resourceName, PEReader assemblyReader, Stream resourceStream)
+        {
+            var metadataReader = assemblyReader.GetMetadataReader();
+
+            bool isResourcePresent = false;
```

Comment:
I think the `isResourcePresent` variable is no longer needed. This method can return `false` if `metadataReader.Resources().TryGetValue` is false and the condition in the second if statement would also be simplified.

Reply:
removed, the try-get is now part of the if-condition